### ISSUE TITLE
The accuracy of DPSGD_Kears only have 10%

### ISSUE THREAD

**Issue by Mrxulovemingming** [tensorflow/privacy#160]
Use DPSGD in the previous mnist_dpsgd_tutorial.py
There can be 90% accuracy in the file, but in mnist_dpsgd_tutorial_keras.py
The medium accuracy is about 10%. I did not modify the relevant code, but the result of DPSGD_keras does not seem to be correct
![image](https://user-images.githubusercontent.com/40209130/112479812-584af500-8db0-11eb-9121-6faeb8c58192.png)

**Comment by fraboeni** [comment]
I observed the same behavior with the VectorizedDPSGD.

**Comment by fraboeni** [comment]
Also tried out the DPKerasSGDOptimizer (in comparison to e.g. the DPGradientDescentGaussianOptimizer) and observed that the initialization might not work properly.
```optimizer = DPKerasSGDOptimizer(
 #optimizer = VectorizedDPSGD(
 #optimizer=DPGradientDescentGaussianOptimizer(
            l2_norm_clip=L2_NORM_CLIP,
            noise_multiplier=NOISE,
            num_microbatches=MICROBATCHES,
            learning_rate=LEARNING_RATE)```

When I had a look into the optimizer object in the debugger after running these lines of code, this is what I obtain in the DPGradientDescentGaussianOptimizer.
![dpgaussian](https://user-images.githubusercontent.com/13484092/113319312-96e13200-9311-11eb-93c0-6bcdc70986b9.png)

And this is for DPKerasSGDOptimizer
![kerassgd](https://user-images.githubusercontent.com/13484092/113319387-a6607b00-9311-11eb-9f0b-489026aa843d.png)

The noise value and clip norm, even though specified seem to be None values.

**Comment by schien1729** [comment]
Thanks for the report-- I'll take a look.

**Comment by schien1729** [comment]
Hi @Mrxulovemingming ,

Can you share your command line? I just tried this command line:

`python3 code/tutorials/mnist_dpsgd_tutorial_keras.py`

with no extra arguments and got this output:

![Screen Shot 2021-04-12 at 10 32 04 PM](https://user-images.githubusercontent.com/45402855/114501946-cd576080-9bdf-11eb-9612-0e8424fa2ee6.png)

**Comment by Mrxulovemingming** [comment]
Hello Schien, I use the same code as you. I run the code in the Spyder environment, and I did not modify anything in the code. As you can see, my running accuracy has been around 10%. I don't know if this is due to a problem with the version of the imported package.

**Comment by Mrxulovemingming** [comment]
![image](https://user-images.githubusercontent.com/40209130/115648901-dfac6980-a358-11eb-8fc2-753c39c1b5a2.png)
This is the result I just ran

**Comment by Mrxulovemingming** [comment]
I used the CPU to run the code and got the same result as you. But the accuracy of running the GPU is still only 10%. What may be the reason for this. The CPU runs very slowly
![Uploading image.png…]()

**Comment by schien1729** [comment]
Thanks for the extra information. I'm looking into it.

**Comment by Mrxulovemingming** [comment]
I have solved this problem,when i using the tensorflow 2.2.0.The accuracy stay at 10%.But when I use the tensorflow 2.4.0 ,cuda 11.1,the result is the same as you
